airflow/contrib/executors/kubernetes_executor.py

```diff
@@ -0,0 +1,236 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import calendar
+import logging
+import time
+
+import multiprocessing
+from airflow.contrib.kubernetes.kubernetes_job_builder import KubernetesJobBuilder
+from airflow.contrib.kubernetes.kubernetes_helper import KubernetesHelper
+from queue import Queue
+from kubernetes import watch
+from airflow import settings
+from airflow.contrib.kubernetes.kubernetes_request_factory import SimpleJobRequestFactory
+from airflow.executors.base_executor import BaseExecutor
+from airflow.models import TaskInstance
+from airflow.utils.state import State
+import json
+# TODO this is just for proof of concept. remove before merging.
+
+
+
+def _prep_command_for_container(command):
+    """  
+    When creating a kubernetes pod, the yaml expects the command
+    in the form of ["cmd","arg","arg","arg"...]
+    This function splits the command string into tokens 
+    and then matches it to the convention.
+
+    :param command:
+
+    :return:
+
+    """
+    return '"' + '","'.join(command.split(' ')[1:]) + '"'
+
+
+class KubernetesJobWatcher(multiprocessing.Process, object):
+    def __init__(self,namespace, np, result_queue, watcher_queue):
+        multiprocessing.Process.__init__(self)
+        self.result_queue = result_queue
+        self._watch_function = namespace
+        self._watch = watch.Watch()
+        self.namespace = np
+        self.watcher_queue = watcher_queue
+
+    def run(self):
+        logging.info("starting watch!")
```

Comment:
Please properly initialise the logger (logging.getLogger(__name__) ) - And I know we don't do this at other places, yet.